controllers/accountclaim/accountclaim_controller.go

```diff
@@ -146,20 +147,34 @@ func (r *AccountClaimReconciler) Reconcile(ctx context.Context, request ctrl.Req
 		return reconcile.Result{}, r.statusUpdate(reqLogger, accountClaim)
 	}
 
-	accountList := &awsv1alpha1.AccountList{}
-
-	listOpts := []client.ListOption{
-		client.InNamespace(awsv1alpha1.AccountCrNamespace),
-	}
-	if err = r.Client.List(context.TODO(), accountList, listOpts...); err != nil {
-		reqLogger.Error(err, "Unable to get accountList")
-		return reconcile.Result{}, err
-	}
-
 	var unclaimedAccount *awsv1alpha1.Account
 
 	// Get an unclaimed account from the pool
 	if accountClaim.Spec.AccountLink == "" {
+
+		accountList := &awsv1alpha1.AccountList{}
+
+		listOpts := []client.ListOption{
+			client.InNamespace(awsv1alpha1.AccountCrNamespace),
+		}
+
+		if err = r.Client.List(context.TODO(), accountList, listOpts...); err != nil {
+			reqLogger.Error(err, "Unable to get accountList")
+			return reconcile.Result{}, err
+		}
+
+		// If the AccountClaim is targetting a specific AccountPool, we want to claim an account from it
+
+		filteredAccountList := &awsv1alpha1.AccountList{}
```

Comment:
I think we'll need to filter _out_ accounts from the non-default accountpool if it's not defined, too, right?

Otherwise there's a chance that a normal OSD account will attempt to claim one from the other pool.